From edb0d2f7e19d28c84040ce967d82add215bb1092 Mon Sep 17 00:00:00 2001
From: Arian <arian.kulmer@web.de>
Date: Tue, 15 Oct 2024 01:44:04 +0200
Subject: [PATCH 9/9] Start gallery as a new task

When launched from lockscreen user will have to unlock for the gallery to be opened, then the gallery is living in the camera process which is wrong.
Let it create a new task for it.
---
 smali/com/android/camera/l3.smali | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/smali/com/android/camera/l3.smali b/smali/com/android/camera/l3.smali
index 81652eeaa..fd0988934 100644
--- a/smali/com/android/camera/l3.smali
+++ b/smali/com/android/camera/l3.smali
@@ -1061,6 +1061,10 @@
 
     invoke-virtual {p0, v0}, Landroid/content/Intent;->addCategory(Ljava/lang/String;)Landroid/content/Intent;
 
+    # FLAG_ACTIVITY_NEW_TASK
+    const v0, 0x10000000
+    invoke-virtual {p0, v0}, Landroid/content/Intent;->addFlags(I)Landroid/content/Intent;
+
     invoke-virtual {p1, p0}, Landroid/app/Activity;->startActivity(Landroid/content/Intent;)V
 
     invoke-virtual {p1}, Lcom/android/camera/ActivityBase;->ri()V
-- 
2.34.1

